### PR TITLE
SW-4578 / SW-4646 Hide sublocations when there are no selections

### PR DESCRIPTION
**Pull request by karthikbtf**
- also update sublocations when the selected facility id changes
- clear selected sublocations when sublocations change

screencast 2024-01-03 10-21-28

### CODE REVIEW

File: src/components/InventoryV2/form/useSubLocations.ts [hunk 1]
```diff
@@ -37,10 +37,8 @@ export const useSubLocations = (nurseryId?: number, record?: { subLocationIds?:
   }, [availableSubLocations, record?.subLocationIds]);
 
   useEffect(() => {
-    if (!availableSubLocations) {
-      void initSubLocations();
-    }
-  }, [availableSubLocations, initSubLocations]);
+    void initSubLocations();
```
Comment (karthikbtf) [inline src/components/InventoryV2/form/useSubLocations.ts:40]: initSublocations callback changes when nursery id or record's facility id changes, we need to refetch sublocations when either of those change

File: src/components/InventoryV2/form/BatchDetailsForm.tsx [hunk 2]
```diff
@@ -230,6 +230,24 @@ export default function BatchDetailsForm(props: BatchDetailsModalProps): JSX.Ele
...
+      return {
+        ...previousRecord,
+        subLocationIds: [],
+      };
+    });
+  }, [availableSubLocations, record?.id, setRecord]);
+
   return (
     <>
       {record && (
```
Comment (karthikbtf) [inline src/components/InventoryV2/form/BatchDetailsForm.tsx:253]: @nickgraz would you know how I can trigger this read-only view? can't seem to find the use-case in the inventory views.

Comment (nickgraz) [reply src/components/InventoryV2/form/BatchDetailsForm.tsx:253]: It used to be the main way to view a batch's details, but now they are forwarded to the batch details page. I think you can still trigger it if you have a q

Comment (nickgraz) [reply src/components/InventoryV2/form/BatchDetailsForm.tsx:253]: It used to be the main way to view a batch's details, but now they are forwarded to the batch details page. I think you can still trigger it if you have a query param in the parent view for the open batch. I suspect there are still other views out there that might link to it that way.

**Review by nickgraz — APPROVED** [review]
Looks good to me

**Pull request merged**